spec/models/card_spec.rb

```diff
@@ -1,33 +1,103 @@
-    let(:card) { Card.new(original_text: 'Good', translated_text: 'GoOd') }
+    context '#words_cannot_be_equal' do
 
-    it 'does not pass the same words' do
-      expect(card).not_to be_valid
+      it 'does not pass the same words' do
+        expect(wrong_card).not_to be_valid
+      end
     end
-  end
 
-  context '#add_review_date' do
-    it 'equals time now + 3days' do
-      expect(card.review_date).to eq(Date.today + 3.days)
+    context '#prepare_word' do
+      it 'work with cyrillic' do
+        expect(wrong_card).not_to be_valid
+      end
     end
-  end
 
-  context '#prepare_word' do
-    it 'work with cyrillic' do
-      expect(card).not_to be_valid
+    context '#check_answer' do
+      context 'when answer is correct' do
+        let!(:correct_answer) { card.check_answer('Hello') }
+
+        it 'user\'s answer equals to original text' do
+          expect(correct_answer).to be true
+        end
+
+        it 'increment field correct_answers by 1' do
+          expect(card.correct_answers).to eq 1
+        end
+      end
+
+      context 'when answer is incorrect' do
+        let!(:incorrect_answer) { card.check_answer('Aloha') }
+
+        it 'user\'s answer not equals to original text' do
+          expect(incorrect_answer).to be false
+        end
+
+        it 'increment field incorrect_answers by 1' do
+          expect(card.incorrect_answers).to eq 1
+        end
+      end
+    end
+
+    before do
+      card.save
     end
-  end
 
-  context '#check_answer' do
-    let(:answer) { 'Привет' }
+    context '#set_review_interval' do
+      context 'when 1 correct answer given' do
+        before do
+          card.check_answer(card.original_text)
+        end
+
+        it 'sets review date +12 hours from now' do
+          expect(card.review_date).to eq (Time.now + 12.hours).to_date
+        end
+
+        context 'and then gives one more correct answer' do
+          it 'set review date +3 days from now' do
+            card.check_answer(card.original_text)
+            expect(card.review_date).to eq (Time.now + 3.days).to_date
+          end
+        end
+
+        context 'and then give 3 incorrect answers' do
+          it 'set review date +12 hours from now' do
+            3.times { card.check_answer(card.translated_text) }
+            expect(card.review_date).to eq (Time.now + 12.hours).to_date
+          end
+        end
+      end
+
+      context 'when gives 3 times correct answer' do
+        it 'set review date +1 week from now ' do
+          3.times { card.check_answer(card.original_text) }
+          expect(card.review_date).to eq (Time.now + 1.week).to_date
+        end
+      end
+
+      context 'when gives 4 times correct answer' do
+        it 'set review date +2 weeks from now ' do
+          4.times { card.check_answer(card.original_text) }
+          expect(card.review_date).to eq (Time.now + 2.week).to_date
+        end
+      end
+
+      context 'when gives 5 times correct answer' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.